x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/DocumentPermissions.java

```diff
@@ -216,14 +217,11 @@ static void failIfQueryUsesClient(QueryBuilder queryBuilder, QueryRewriteContext
      * @return {@link DocumentPermissions}
      */
     public static DocumentPermissions filteredBy(Set<BytesReference> queries) {
-        if (queries == null || queries.isEmpty()) {
-            throw new IllegalArgumentException("null or empty queries not permitted");
-        }
         return new DocumentPermissions(queries);
     }
 
     /**
-     * Create {@link DocumentPermissions} with no restriction. The {@link #getQueries()}
+     * Create {@link DocumentPermissions} with no restriction. The {@link #getListOfQueries()}
```

Comment:
not related to the change here ...but i think this javadoc is wrong, would you mind nuking it ?

Reply:
Why do you think it is wrong? The word `Create` is not particularly accurate because the method always returns the same instance. But otherwise it is still accurate?

Reply:
The javadoc is not really for this method since it describes how other methods work. It's not a big deal, just confusing. 

IMO the javadoc should read
```
Returns a {@link DocumentPermissions} with no restrictions. This is equivalent to not having any document level permissions. 
```
(or just not exist since it kinda self evident)
Also, outside the scope of this change, so feel free to disregard this comment.

Reply:
Ah ok. Thanks for explaining this further. I see your point now. I removed the javadoc since it is indeed self evident.